code/game/machinery/doors/door.dm

```diff
@@ -149,26 +149,36 @@
 				var/mob/living/simple_animal/bot/B = user
 				B.door_opened(src)
 		else
+			if(pry_open_check(user))
+				return
 			do_animate("deny")
-			if(HAS_TRAIT(user, TRAIT_FORCE_DOORS))
-				var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
-
-				if(V && HAS_TRAIT_FROM(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT))
-					if(!V.bloodusable)
-						REMOVE_TRAIT(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT)
-						return
-				if(welded)
-					to_chat(user, "<span class='warning'>The door is welded.</span>")
-					return
-				if(locked)
-					to_chat(user, "<span class='warning'>The door is bolted.</span>")
-					return
-				if(density)
-					visible_message("<span class='danger'>[user] forces the door open!</span>")
-					playsound(loc, "sparks", 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
-					open(TRUE)
-				if(V && HAS_TRAIT_FROM(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT))
-					V.bloodusable = max(V.bloodusable - 5, 0)
+
+/obj/machinery/door/proc/pry_open_check(mob/user)
+	. = TRUE
+	if(isterrorspider(user))
+		return
+
+	if(HAS_TRAIT(user, TRAIT_FORCE_DOORS))
+		var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
+
+		if(V && HAS_TRAIT_FROM(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT))
+			if(!V.bloodusable)
+				REMOVE_TRAIT(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT)
+				return FALSE
+		if(welded)
+			to_chat(user, "<span class='warning'>The door is welded.</span>")
+			return FALSE
+		if(locked)
+			to_chat(user, "<span class='warning'>The door is bolted.</span>")
+			return FALSE
+		if(density)
+			visible_message("<span class='danger'>[user] forces the door open!</span>")
+			playsound(loc, "sparks", 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+			open(TRUE)
+		if(V && HAS_TRAIT_FROM(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT))
+			V.bloodusable = max(V.bloodusable - 5, 0)
+		return
+	return FALSE
```

Comment:
```suggestion
	if(!HAS_TRAIT(user, TRAIT_FORCE_DOORS))
		return FALSE
	var/datum/antagonist/vampire/V = user.mind?.has_antag_datum(/datum/antagonist/vampire)

	if(V && HAS_TRAIT_FROM(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT))
		if(!V.bloodusable)
			REMOVE_TRAIT(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT)
			return FALSE
	if(welded)
		to_chat(user, "<span class='warning'>The door is welded.</span>")
		return FALSE
	if(locked)
		to_chat(user, "<span class='warning'>The door is bolted.</span>")
		return FALSE
	if(density)
		visible_message("<span class='danger'>[user] forces the door open!</span>")
		playsound(loc, "sparks", 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
		open(TRUE)
	if(V && HAS_TRAIT_FROM(user, TRAIT_FORCE_DOORS, VAMPIRE_TRAIT))
		V.bloodusable = max(V.bloodusable - 5, 0)
```
Early return is preferred. And to fix a runtime where a mindless mob bumps into a door